lib/common.sh

```diff
@@ -213,6 +229,21 @@ resolve_ipv4()
         return 1
     fi
 
+    #
+    # Check if the IP-FQDN pair is present in /etc/hosts
+    # 
+    if is_present_in_etc_hosts "$ipv4_addr" "$hname"; then
+        if [ "$fail_if_present_in_etc_hosts" == "true" ]; then
+            eecho "[FATAL] Detected entry $ipv4_addr $hname in /etc/hosts."
+            eecho "$hname is resolving into $ipv4_addr from /etc/hosts."
+            eecho "Remove the entry for "$hname" in /etc/hosts to ensure proper handling of endpoint IP address changes by AZNFS."
```

Comment:
eecho "[FATAL] $hname resolved to $ipv4_addr from /etc/hosts!"
 eecho "AZNFS depends on dynamically detecting DNS changes for proper handling of endpoint address changes"
 eecho "Please remove the entry for $hname from /etc/hosts"